server/channels/app/imports/import_validators.go

```diff
@@ -775,19 +789,26 @@ func isValidGuestRoles(data UserImportData) bool {
 	return true
 }
 
-// ValidateAttachmentPathForImport checks if the provided path is valid for import.
-// It ensures that the path is cleaned and resolved against a base path.
-// It returns the resolved path and a boolean indicating if the path is valid.
+// ValidateAttachmentPathForImport joins 'path' to 'basePath' (defaulting to "." if empty) and ensures
+// the result does not escape the base directory. Returns the cleaned joined path (and true),
+// or an empty string (and false) if the result escapes the base.
 func ValidateAttachmentPathForImport(path, basePath string) (string, bool) {
-	cleanedPath := filepath.Clean(path)
-	resolvedPath := filepath.Join(basePath, cleanedPath)
+	if basePath == "" {
+		basePath = "."
+	}
 
-	// Check if the resolved path is within the expected base path.
-	if !strings.HasPrefix(resolvedPath, basePath) {
+	joined := filepath.Join(basePath, path)
+
+	// Check if the resolved joined path is within basePath
+	rel, err := filepath.Rel(basePath, joined)
+	if err != nil {
+		return "", false
+	}
+	if strings.HasPrefix(rel, ".."+string(filepath.Separator)) || rel == ".." {
 		return "", false
 	}
```

Comment:
This seems correct, but oddly counter intuitive. The comment would have to be something like, "Check if the relative path between `basePath` and `joined` requires accessing the parent directory, thus implying the joined path is not within `basePath`." I /think/ that works, but I admit I'm not confident.

Acknowledging the discussed limitations of `strings.HasPrefix`, would it make sense to just check 
```go
// Ensure basePath ends with separator for safer prefix checking
if !strings.HasSuffix(basePath, string(filepath.Separator)) {
    basePath = basePath + string(filepath.Separator)
}

// Old check...
if !strings.HasPrefix(resolvedPath, basePath)
```